tools/hsmtool.c

```diff
@@ -319,5 +392,16 @@ int main(int argc, char *argv[])
 		                              argv[5], argv[6]);
 	}
 
+	if (streq(method, "guesstoremote")) {
+		/*   address    node_id    depth    hsm_secret  ?password? */
+		if (!(argv[2] && argv[3] && argv[4] && argv[5]))
+			show_usage();
```

Comment:
This will likely upset valgrind. Better to check `argc` before attempting to access the array elements.